tests/common/hashkv_test.go

```diff
@@ -0,0 +1,248 @@
+// Copyright 2025 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"go.etcd.io/etcd/tests/v3/framework/config"
+	intf "go.etcd.io/etcd/tests/v3/framework/interfaces"
+	"go.etcd.io/etcd/tests/v3/framework/testutils"
+)
+
+// TestVerifyHashKVAfterCompact tests that HashKV is consistent across all members after a physical compaction.
+// It tests both cases where the compaction is on a tombstone revision and where it is not.
+func TestVerifyHashKVAfterCompact(t *testing.T) {
```

Comment:
Migrate this test:
https://github.com/etcd-io/etcd/blob/991ea7e0b4762303878f5b3c0cb4edc68c16bd39/tests/e2e/hashkv_test.go#L31